react/src/components/ComputeSessionNodeItems/TerminateSessionModal.tsx

```diff
@@ -0,0 +1,381 @@
+    baiClient,
+  );
+
+  const rqst = {
+    method: 'GET',
+    uri: new URL(`proxy/${accessKey}/${session.row_id}`, proxyURL).href,
+  };
+
+  return sendRequest(rqst).then((response) => {
+    let uri = new URL(`proxy/${accessKey}/${session.row_id}/delete`, proxyURL);
+    if (localStorage.getItem('backendaiwebui.appproxy-permit-key')) {
+      uri.searchParams.set(
+        'permit_key',
+        localStorage.getItem('backendaiwebui.appproxy-permit-key') || '',
+      );
+      uri = new URL(uri.href);
+    }
+    if (response !== undefined && response.code !== 404) {
+      return sendRequest({
+        method: 'GET',
+        uri: uri.href,
+        credentials: 'include',
+        mode: 'cors',
+      });
+    }
+    return true;
+  });
+};
+
+const TerminateSessionModal: React.FC<TerminateSessionModalProps> = ({
+  sessionFrgmts: sessionFrgmt,
+  onRequestClose,
+  ...modalProps
+}) => {
+  const openTerminateModal = false;
+  const { t } = useTranslation();
+  const { styles } = useStyle();
+  const sessions = useFragment(
+    graphql`
+      fragment TerminateSessionModalFragment on ComputeSessionNode
+      @relay(plural: true) {
+        id
+        row_id
+        name
+        scaling_group @required(action: NONE)
+        kernel_nodes(first: 100) {
+          edges {
+            node {
+              container_id
+              agent_id
+            }
+          }
+        }
+      }
+    `,
+    sessionFrgmt,
+  );
+  const [isForce, setIsForce] = useState(false);
+  const userRole = useCurrentUserRole();
+
+  const baiClient = useSuspendedBackendaiClient();
+
+  const currentProject = useCurrentProjectValue();
+
+  const terminateMutation = useTanMutation({
+    mutationFn: async (session: Session) => {
+      return terminateApp(
+        session,
+        baiClient._config.accessKey,
+        currentProject.id,
+        baiClient,
+      )
+        .catch((e) => {
+          return {
+            error: e,
+          };
+        })
+        .then((result) => {
+          const err = result?.error;
+          if (
+            err === undefined || //no error
+            (err && // Even if wsproxy address is invalid, session must be deleted.
+              err.message &&
+              (err.statusCode === 404 || err.statusCode === 500))
+          ) {
+            // BAI client destroy try to request 3times as default
+            return baiClient.destroy(
+              session.row_id,
+              baiClient._config.accessKey,
+              isForce,
+            );
+          } else {
+            throw err;
+          }
+        });
+    },
+  });
+  const relayEvn = useRelayEnvironment();
+  const painKiller = usePainKiller();
+  const { upsertNotification } = useSetBAINotification();
+
+  return (
+    <BAIModal
+      centered
+      title={t('session.TerminateSession')}
+      open={openTerminateModal}
+      confirmLoading={terminateMutation.isPending}
+      onOk={() => {
+        if (sessions[0]?.row_id) {
```

Comment:
Doesn't it currently offer the ability to end multiple sessions at once?

Reply:
Yes not now. Let's update it with WebUI NEO session list.